core/src/test/java/org/mobilitydata/gtfsvalidator/table/AnyTableLoaderTest.java

```diff
@@ -0,0 +1,239 @@
+    assertThat(loaderNotices.getValidationNotices())
+        .containsExactly(new EmptyFileNotice("filename"));
+    assertThat(loadedContainer).isEqualTo(mockContainer);
+  }
+
+  @Test
+  public void invalidHeaders() {
+    var testTableDescriptor = mock(GtfsTableDescriptor.class);
+    when(testTableDescriptor.gtfsFilename()).thenReturn("filename");
+    when(testTableDescriptor.getColumns())
+        .thenReturn(
+            ImmutableList.of(
+                GtfsColumnDescriptor.builder()
+                    .setColumnName("route_id")
+                    .setHeaderRequired(true)
+                    .setFieldLevel(FieldLevelEnum.REQUIRED)
+                    .setIsMixedCase(false)
+                    .setIsCached(false)
+                    .build()));
+    GtfsTableContainer mockContainer = mock(GtfsTableContainer.class);
+    when(testTableDescriptor.createContainerForInvalidStatus(
+            GtfsTableContainer.TableStatus.INVALID_HEADERS))
+        .thenReturn(mockContainer);
+    ValidatorProvider validatorProvider =
+        spy(
+            new DefaultValidatorProvider(
+                mock(ValidationContext.class), mock(ValidatorLoader.class)));
+    NoticeContainer loaderNotices = new NoticeContainer();
+    InputStream csvInputStream = toInputStream("A file with no headers");
+
+    var loadedContainer =
+        AnyTableLoader.load(testTableDescriptor, validatorProvider, csvInputStream, loaderNotices);
+
+    assertThat(loaderNotices.hasValidationErrors()).isTrue();
+    assertThat(loaderNotices.getValidationNotices().size()).isEqualTo(2);
+    assertThat(loaderNotices.getValidationNotices())
+        .contains(new UnknownColumnNotice("filename", "A file with no headers", 1));
+    assertThat(loaderNotices.getValidationNotices())
+        .contains(new MissingRequiredColumnNotice("filename", "route_id"));
+    assertThat(loadedContainer).isEqualTo(mockContainer);
+  }
+
+  @Test
+  public void invalidRowLengthNotice() {
+    var testTableDescriptor = spy(new GtfsTestTableDescriptor());
+    GtfsTableContainer mockContainer = mock(GtfsTableContainer.class);
+    when(testTableDescriptor.createContainerForInvalidStatus(
+            GtfsTableContainer.TableStatus.UNPARSABLE_ROWS))
+        .thenReturn(mockContainer);
+    ValidatorProvider validatorProvider =
+        spy(new DefaultValidatorProvider(validationContext, ValidatorLoader.createEmpty()));
+    NoticeContainer loaderNotices = new NoticeContainer();
+    InputStream inputStream = toInputStream("id,code\n" + "s1\n");
+
+    var loadedContainer =
+        AnyTableLoader.load(testTableDescriptor, validatorProvider, inputStream, loaderNotices);
+
+    assertThat(loaderNotices.hasValidationErrors()).isTrue();
+    assertThat(loaderNotices.getValidationNotices())
+        .containsExactly(new InvalidRowLengthNotice("filename.txt", 2, 1, 2));
+    assertThat(loadedContainer).isEqualTo(mockContainer);
+  }
+
+  @Test
+  public void missingRequiredColumn() {
```

Comment:
This seems like a unit-test of `DefaultTableHeaderValidator` and its presence in DefaultValidatorProvider.  While that may be a reasonable thing to test, does it belong in `AnyTableLoader` unit test?

Reply:
Removed the test in favor of a more generic test _invalidHeaders_. Also, removed all dependencies with DefaultTableHeaderValidator and DefaultValidatorProvider.